dm/loader/loader.go

```diff
@@ -362,7 +362,7 @@ func (w *Worker) dispatchSQL(ctx context.Context, file string, offset int64, tab
 				continue
 			}
 
-			if w.loader.columnMapping != nil {
+			if w.loader.columnMapping != nil || len(table.extendCol) > 0 {
```

Comment:
why we need this check: `len(table.extendCol) > 0` ?

Reply:
If the extend column exists, we need use `reassemble` rewrite SQL.

Reply:
ohoho i see, but there is another case we need to handle:
`` `
table.sourceTable != table.targetTable && len(table.extendCol) > 0
```

Reply:
I think we don't need handle it.

Since `reassemble` use `table.insertHeadStmt` generate SQL, it seems that `dstTable` is always the target table name. Or is there any other reason?

https://github.com/yufan022/ticdc/blob/dm_2111/dm/loader/convert_data.go#L318
https://github.com/yufan022/ticdc/blob/dm_2111/dm/loader/convert_data.go#L297